modules/core/include/opencv2/core/hal/intrin_neon.hpp

```diff
@@ -1212,13 +1212,13 @@ OPENCV_HAL_IMPL_NEON_SHIFT_OP(v_int64x2, s64, int64, s64)
 template<int n> inline _Tpvec v_rotate_right(const _Tpvec& a) \
 { return _Tpvec(vextq_##suffix(a.val, vdupq_n_##suffix(0), n)); } \
 template<int n> inline _Tpvec v_rotate_left(const _Tpvec& a) \
-{ return _Tpvec(vextq_##suffix(vdupq_n_##suffix(0), a.val, _Tpvec::nlanes - n)); } \
+{ return _Tpvec(vextq_##suffix(vdupq_n_##suffix(0), a.val, _Tpvec::max_nlanes - n)); } \
```

Comment:
Why is this using `max_nlanes`? Shouldn't it be using `nlanes`? 

I actually think that in all the `intrin_*.hpp` files the intrinsics defined in there should be using `nlanes`, not `max_nlanes`.

@vpisarev  - am I missing something here?

Reply:
nlanes should be fine, it's likely an artifact from testing